packages/core/src/StateMachine.ts

```diff
@@ -378,15 +393,32 @@ export class StateMachine<
       | StateConfig<TContext, TEvent>
   ): State<TContext, TEvent, TResolvedTypesMeta> {
     const state =
-      stateConfig instanceof State
-        ? stateConfig
-        : (new State(stateConfig) as State<
-            TContext,
-            TEvent,
-            TResolvedTypesMeta
-          >);
-    state.machine = this;
-    return state;
+      stateConfig instanceof State ? stateConfig : new State(stateConfig, this);
+
+    return state as State<TContext, TEvent, TResolvedTypesMeta>;
+  }
+
+  public getStatus(state: State<TContext, TEvent, TResolvedTypesMeta>) {
+    return state.done
+      ? { status: 'done', data: state.output }
+      : { status: 'active' };
+  }
```

Comment:
We probably need an extra `error` status for cases when something throws an error and we "forcefully" stop the machine (and notify subscribers about the `error`)